core/amber/src/main/python/core/models/schema/test_schema.py

```diff
@@ -0,0 +1,74 @@
+import pytest
+import pyarrow as pa
+
+from core.models.schema.attribute_type import AttributeType
+from core.models.schema.schema import Schema
+
+
+class TestSchema:
+    @pytest.fixture
+    def raw_schema(self):
+        return {
+            "field-1": "string",
+            "field-2": "integer",
+            "field-3": "long",
+            "field-4": "double",
+            "field-5": "boolean",
+            "field-6": "timestamp",
+            "field-7": "binary",
+        }
+
+    @pytest.fixture
+    def arrow_schema(self):
+        return pa.schema(
+            [
+                pa.field("field-1", pa.string()),
+                pa.field("field-2", pa.int32()),
+                pa.field("field-3", pa.int64()),
+                pa.field("field-4", pa.float64()),
+                pa.field("field-5", pa.bool_()),
+                pa.field("field-6", pa.timestamp("ms", tz="UTC")),
+                pa.field("field-7", pa.binary()),
+            ]
+        )
+
+    @pytest.fixture
+    def schema(self):
+        s = Schema()
+        s.add("field-1", AttributeType.STRING)
+        s.add("field-2", AttributeType.INT)
+        s.add("field-3", AttributeType.LONG)
+        s.add("field-4", AttributeType.DOUBLE)
+        s.add("field-5", AttributeType.BOOL)
+        s.add("field-6", AttributeType.TIMESTAMP)
+        s.add("field-7", AttributeType.BINARY)
+        return s
+
+    def test_accessors_and_mutators(self, schema):
+        assert schema.get_attr_names() == [f"field-{i}" for i in range(1, 8)]
+        assert schema.get_attr_type("field-2") == AttributeType.INT
+        assert schema.get_attr_type("field-6") == AttributeType.TIMESTAMP
+        assert schema.as_key_value_pairs() == [
+            ("field-1", AttributeType.STRING),
+            ("field-2", AttributeType.INT),
+            ("field-3", AttributeType.LONG),
+            ("field-4", AttributeType.DOUBLE),
+            ("field-5", AttributeType.BOOL),
+            ("field-6", AttributeType.TIMESTAMP),
+            ("field-7", AttributeType.BINARY),
+        ]
+        with pytest.raises(KeyError):
+            schema.get_attr_type("does not exist")
+        with pytest.raises(TypeError):
+            schema["illegal_assign"] = "value"
+        with pytest.raises(TypeError):
+            _ = schema["illegal_access"]
+        with pytest.raises(KeyError):
+            schema.add("field-2", AttributeType.LONG)
+
+    def test_convert_from_raw_schema(self, raw_schema, schema):
```

Comment:
Where will this be used? I cannot find a place that references it.

Reply:
During initialization, Java passes the defined raw schema to Python, which describes the output fields. See this line https://github.com/Texera/texera/blob/cdb8623dfc927befb7faa9b3bfbf4c753e4c6678/core/amber/src/main/python/core/models/operator.py#L52